vyxal/elements.py

```diff
@@ -33,5 +45,471 @@ def process_element(
+        simple = {
+            (SCALAR_TYPE, SCALAR_TYPE): lambda: safe_apply(
+                function, lhs, rhs, ctx=ctx
+            ),
+            (SCALAR_TYPE, list): lambda: (
+                safe_apply(function, lhs, x, ctx=ctx) for x in rhs
+            ),
+            (list, SCALAR_TYPE): lambda: (
+                safe_apply(function, x, rhs, ctx=ctx) for x in lhs
+            ),
+            (list, list): lambda: (
+                safe_apply(function, x, y, ctx=ctx) for x, y in vy_zip(lhs, rhs)
+            ),
+        }
+
+        explicit_dict = {
+            (SCALAR_TYPE, SCALAR_TYPE): lambda: (
+                safe_apply(function, x, rhs, ctx=ctx) for x in iterable(lhs)
+            ),
+            (SCALAR_TYPE, list): lambda: (
+                safe_apply(function, lhs, x, ctx=ctx) for x in rhs
+            ),
+            (list, SCALAR_TYPE): lambda: (
+                safe_apply(function, x, rhs, ctx=ctx) for x in lhs
+            ),
+            (list, list): lambda: (
+                safe_apply(function, x, rhs, ctx=ctx) for x in lhs
+            ),
+        }
+
+        if explicit:
+            return list(explicit_dict.get(ts)())
+        else:
+            return list(simple.get(ts)())
+    else:
+        # That is, single argument vectorisation
+        if explicit:
+            lhs = iterable(lhs, range, ctx=ctx)
+        else:
+            lhs = iterable(lhs, ctx=ctx)
+
+        return LazyList((safe_apply(function, x, ctx=ctx) for x in lhs))
+
+
+def vectorised_not(lhs, ctx):
+    return {NUMBER_TYPE: lambda: int(not lhs), str: lambda: int(not lhs)}.get(
+        vy_type(lhs), lambda: vectorise(vectorised_not, lhs, ctx=ctx)
+    )()
+
+
+def vy_print(lhs, end="\n", ctx=None):
+    """Element ,
+    (any) -> send to stdout
+    """
+
+    ctx.printed = True
+    ts = vy_type(lhs)
+
+    if ts is LazyList:
+        lhs.output(end, ctx)
+    elif ts is list:
+        LazyList(lhs).output(end, ctx)
+    else:
+        if ctx.online:
+            ctx.online_output += str(lhs)
+            # TODO: use custom string func
+        else:
+            print(lhs, end=end)
+
+
+def vy_type(item, other=None, simple=False):
+    if other is not None:
+        return (vy_type(item, simple=simple), vy_type(other, simple=simple))
+    if (x := type(item)) in (int, sympy.Rational, complex):
+        return NUMBER_TYPE
+    elif simple and isinstance(item, LazyList):
+        return list
+    else:
+        return x
+
+
+elements: dict[str, tuple[str, int]] = {
+    "¬": process_element("int(not lhs)", 1),
+    "∧": process_element("lhs and rhs", 2),
+    "⟑": process_element("rhs and lhs", 2),
+    "∨": process_element("lhs or rhs", 2),
+    "⟇": process_element("rhs or lhs", 2),
+    "÷": (
+        "lhs = pop(stack, 1, ctx); stack += iterable(lhs, ctx=ctx)",
+        1,
+    ),
+    "×": process_element("'*'", 0),
+    "•": process_element(log_mold_multi, 2),
+    "†": (
+        "top = function_call(stack, ctx)\n"
```

Comment:
Should this be extracted to a proper function? Code inside strings makes me queasy.

Reply:
I feel like this is better than inside a function.